sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala

```diff
@@ -258,7 +258,7 @@ package object expressions  {
         case (Seq(), _) =>
           val name = nameParts.head
           val attributes = collectMatches(name, direct.get(name.toLowerCase(Locale.ROOT)))
-          (attributes.filterNot(_.qualifiedAccessOnly), nameParts.tail)
+          (attributes.distinct.filterNot(_.qualifiedAccessOnly), nameParts.tail)
```

Comment:
shall we fix `def unique` in this class? It should look at expr Id.

Reply:
The unique method is not used in this flow.  It's used at many places while returning the result. Making any changes to unique will increase the scope.